packages/cloud_firestore/cloud_firestore_web/lib/src/utils/web_utils.dart

```diff
@@ -123,3 +126,51 @@ firestore_interop.SetOptions? convertSetOptions(SetOptions? options) {
 firestore_interop.FieldPath convertFieldPath(FieldPath fieldPath) {
   return firestore_interop.FieldPath(fieldPath.components.toList().join('.'));
 }
+
+FirebaseException buildFirebaseException(
+    core_interop.FirebaseError firebaseError) {
+  String code = firebaseError.code.replaceFirst('firestore/', '');
+  String message =
+      firebaseError.message.replaceFirst('(${firebaseError.code})', '');
+  return FirebaseException(
+      plugin: 'cloud_firestore', code: code, message: message);
+}
+
+/// Will return a [FirebaseException] from a thrown web error.
+/// Any other errors will be propagated as normal.
+Future<R> guard<R>(Future<R> Function() cb) async {
+  try {
+    final value = await cb();
+    return value;
+  } catch (error) {
+    if (error is! core_interop.FirebaseError) {
+      rethrow;
+    }
+
+    throw buildFirebaseException(error);
+  }
+}
+
+R guardSync<R>(R Function() cb) {
+  try {
+    final value = cb();
+    assert(value != Future);
+
+    return cb();
+  } catch (error) {
+    if (error is! core_interop.FirebaseError) {
+      rethrow;
+    }
+
+    throw buildFirebaseException(error);
+  }
+}
+
+/// Returns a [FirebaseException] from a thrown web error.
+Exception getFirebaseException(Object object) {
```

Comment:
Is this still used?

Reply:
Yeah, it is still used. I wasn't sure how to capture the error in error handlers like this one: https://github.com/FirebaseExtended/flutterfire/blob/@russell/firestore-web-exception/packages/cloud_firestore/cloud_firestore_web/lib/src/query_web.dart#L158